### PR TITLE
add grain boundary workflow and tests

### PR DESCRIPTION
**Pull request by Tinaatucsd**
## Summary

If a bulk_structure and grain boundary (GB) generation parameters are specified, the workflow will relax the bulk structure first, then generate GB based on given parameters and TransmuterFW, then relax the GB structure. 

If a grain boundary structure is given, then the workflow will directly run the relaxation for the given GB structure, while the related additional info could be added into the database when the user specified.

### CODE REVIEW

**Comment by computron** [comment]
Hi @Tinaatucsd 

Thanks for these changes; I made some minor comments, please review them and let me know when ready for re-review.

File: atomate/vasp/workflows/base/grain_boundary.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    Function to generate a gb firework. If bulk structure is given, then gb_gen_params is needed to
+    generate corresponding gb structure and a TransmuterFW is generated.
+    If the gb structure is given, an OptimizeFW is returned.
+
+    Args:
+        gb (Structure, Gb): grain boundary structure to be calculated.
+        bulk_structure (Structure): bulk structure, if provided a TransmuterFW will be generated using
+          the necessary transformations to get the gb from the bulk structure. Bulk structure could
+          be primitive cell or conventional unit cell.
+        gb_gen_params (dict): dictionary of gb generation parameters, used to generate the gb,
```
[the remark this answers is not in the document]
Comment (Tinaatucsd) [reply atomate/vasp/workflows/base/grain_boundary.py]: agree, I will refer to the corresponding method.

File: atomate/vasp/workflows/base/grain_boundary.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+          which is necessary to get the gb that corresponds to the bulk structure. One example is
+          given below:
+              gb_s5_001_params = {"rotation_axis": [1, 0, 0], "rotation_angle": 36.86989764584402,
+                        "expand_times": 2, "vacuum_thickness": 0.0, "normal": True,
+                        "ratio": None, "plane": [1, 0, 0]}
+        gb_structure_info (dict): the Gb object stored as dict which include the info of grain boundary
```
[the remark this answers is not in the document]
Comment (Tinaatucsd) [reply atomate/vasp/workflows/base/grain_boundary.py]: Sorry, I forgot to delete gb_structure_info, which was designed to add more info to describe the structure previously, but later on, I notice that I can use add_additional_fields_to_taskdocs to add this info. I will delete gb_structure_info it later.

File: atomate/vasp/workflows/base/grain_boundary.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        vasp_cmd (str): vasp command
+        name (str): Name for the Firework
+
+    Return:
+        Firework
+    """
+    # If a bulk_structure is specified, generate the set of transformations, else
+    # just create an optimize FW for gb
+
+    if bulk_structure:
```
[the remark this answers is not in the document]
Comment (Tinaatucsd) [reply atomate/vasp/workflows/base/grain_boundary.py]: sounds good and easier for users. I will do it.

**Comment by Tinaatucsd** [comment]
Hi @computron,
Thank you for the comments, which will help users to start easily. 
I will change the code accordingly. Hopefully, I can get it done later today, or at least within this week.

**Comment by Tinaatucsd** [comment]
Hi Dr. Anubhav,
I run into some directory problems when I test the code (maybe due to my test configuration). I might need a couple of days to fix the problem. Sorry for the delay, I will try my best to fix it within this week if the time schedule works for you. Thank you. 
Best,
Hui

**Comment by computron** [comment]
Sure, just let me know when you want a re-review of the code

**Comment by computron** [comment]
Hi @Tinaatucsd 

I would suggest that you close this PR and re-open when ready

**Comment by Tinaatucsd** [comment]
Sure, I will close now and re-open a new one when I am ready. Thank you!

**Pull request closed without merging**